.circleci/main.yml

```diff
@@ -684,7 +684,8 @@ jobs:
           command: go install golang.org/x/vuln/cmd/govulncheck@latest
       - run:
           name: Check for known security vulnerabilities
-          command: govulncheck ./...
+          command: |
+            find -name go.mod -print0 | xargs -0 -I{} sh -c 'cd $(dirname {}); echo; echo '---'; echo $PWD; exec govulncheck ./...'
```

Comment:
I don't think either `"$(dirname {})"` or `$(dirname "{}")` fix this issue.

Reply:
You’re right. It needs to be … wait for it … `"$(dirname "{}")"`:

```shell
$ mkdir -p /tmp/test-spaces/foo\ bar/baz/quux
~$ find /tmp/test-spaces -name quux -print
/tmp/test-spaces/foo bar/baz/quux
$ find /tmp/test-spaces -name quux -print0 | xargs -0 -I{} sh -c 'cd "$(dirname "{}")"; pwd'
/tmp/test-spaces/foo bar/baz
```

Single quotes to pass a complete string as an argument to `sh`, the double quotes to pass a complete string to `cd`, then dollar-parentheses for command substitution, then double-quotes to pass a complete string to `dirname`.

Oof, shell can be ‘fun.’

Reply:
Oh, you could also use `printf "\\n---$PWD"` instead of `echo; echo '---'; echo $PWD`. `printf` is [in POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/printf.html), so should be available everywhere.